### ISSUE TITLE
Link widget to a different backend?

### ISSUE THREAD

**Issue by paulowoicho** [catogonzalez/universal-chat-widget#1]
I would like to use this chat widget and connect it to a bot creator that a friend worked on called Botify. The app can be accessed here _botify-demo-9615.herokuapp.com_. This is the github repo _https://github.com/enaiho/botify_

**Comment by datanatas** [comment]
Sure @aspiringDevPaul: connecting the widget to other backend is fairly easy. You can start to develop a new "Adapter" based on one of the two I made for [Rails Action Cable](https://github.com/catogonzalez/chat-adapter-action-cable) or [Rocket Chat](https://github.com/catogonzalez/chat-adapter-rocketchat). The interface is very basic and the code on any of those adapters should be easy to follow. 

I am planning to write more documentation on how to develop the adapters but meanwhile, feel free to drop me a line/comment if you need help.

**Comment by paulowoicho** [comment]
Sorry @catogonzalez, I have one more question. Can this widget be easily embedded in websites that our end users create?

**Comment by datanatas** [comment]
Yes, running `npm run build` will create a `./dist` folder with the css and js files ready for production. Your clients will need to reference those files in their website(s).

**Comment by paulowoicho** [comment]
Hello @catogonzalez, the adapter code is a bit hard to follow for me as a newbie programmer. I was wondering if you could help with some instructions as to how to get started. I could work with the any documentation you have even if it is not yet complete. Thanks so much in advance.

**Comment by datanatas** [comment]
I have to put together some documentation: will open some space to get that done this week. In the meantime, what would be the backend? There is usually some flavor of [websockets](https://en.wikipedia.org/wiki/WebSocket) involved: do you know what flavor you will use?

Here is an article that might help you get started with the tools involved if your backend is node.js: https://medium.com/@martin.sikora/node-js-websocket-simple-chat-tutorial-2def3a841b61

**Comment by paulowoicho** [comment]
The backend is a bot creation tool called Botify. I will be using socket.io for this project

**Comment by paulowoicho** [comment]
Thank you in advance for taking time to do the documentation.

**Comment by datanatas** [comment]
Hi @aspiringDevPaul. I am sorry it took longer than I expected. I have created [this wiki page](https://github.com/catogonzalez/universal-chat-widget/wiki/Creating-new-adapters-to-connect-to-a-different-backend) detailing the interface needed to create the new adapters. I hope you can get it working. Let me know if you need anything else.
